src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/ToolModeSwitchBehavior.java

```diff
@@ -51,6 +72,28 @@ public void addBehaviorNBT(@NotNull ItemStack stack, @NotNull CompoundTag tag) {
                         WrenchModeType.values()[tagCompound.getByte("Mode")].getName()), true);
             }
             return InteractionResultHolder.success(itemStack);
+        } else if (blockHitResult.getType() == HitResult.Type.BLOCK) {
+            BlockState state = world.getBlockState(blockHitResult.getBlockPos());
+            BlockPos pos = blockHitResult.getBlockPos();
+            Set<GTToolType> toolTypes = ToolHelper.getToolTypes(itemStack);
+            if (toolTypes.contains(GTToolType.WRENCH) && GTCEu.Mods.isCreateLoaded() &&
+                    AllTags.AllBlockTags.WRENCH_PICKUP.matches(state)) {
+                if (!(world instanceof ServerLevel))
```

Comment:
Use a pattern variable (and replace all the casts)

```suggestion
                if (!(world instanceof ServerLevel serverLevel))
```